### ISSUE TITLE
The user account profile page has no direct path to link to the 2FA settings page

### ISSUE THREAD

**Issue by fourthletter** [OpenTechFund/apply-app#31]
## Describe the bug

As a user my account profile should have an option for me to edit my 2FA settings without knowing the absolute url.

As a non-SSO user, I could enable 2FA during sign up but there is no option for me to disable and enable 2fa without using the absolute path. There is no link from my account profile to those settings. 

## Expected behavior

Enhance usability for non-SSO users to enable or disable 2FA, and other access other security settings on hypha.

**Priority**
- Medium (slowing you down)

**Affected roles**
- Staff

**Comment by Techslammer** [comment]
User with 2FA enabled logs in with username and password:

![Screenshot 2021-12-07 11 20 52 AM - Display 1](https://user-images.githubusercontent.com/93009150/145086739-295c3345-1308-4b0b-baa0-8a296eeb8bca.png)

User with 2FA inputs their token and is able to be authenticated:

![Screenshot 2021-12-07 11 25 05 AM](https://user-images.githubusercontent.com/93009150/145086799-36be6f89-deb5-434f-a012-705befb507cc.png)

User wants to change their 2fa settings. They go to their account profile. There is no pathway (link, button, etc) to change the settings without using the absolute path to get to the page(https://apply.opentech.fund/account/two_factor/):

![Screenshot 2021-12-07 11 27 06 AM](https://user-images.githubusercontent.com/93009150/145086897-6fa153bb-80cd-41c6-a9f2-2d76bde0b638.png)

**Comment by frjo** [comment]
Is this not a duplicate of https://github.com/OpenTechFund/apply-app/issues/30?

**Comment by Techslammer** [comment]
It's not a duplicate of #30 @frjo  .  I logged in as a non-SSO user and not with SSO.  In this case, i enabled 2FA when I signed up but there is no option for me to disable and enable 2fa without using the absolute path. For example, there is no link from my account profile to those settings.  So if are user that didn't enable 2fa at sign-up, it would be difficult to find the page to enable it later.

**Comment by frjo** [comment]
This is how it looks for normal users.

![Skärmavbild 2021-12-07 kl  22 39 55](https://user-images.githubusercontent.com/191392/145110357-6be88d95-78c3-47b5-90f2-a7d7b25a4f2d.png)

**Comment by fourthletter** [comment]
We could carve out time this upcoming year for a usability review of security settings. Increasing the usability would help ensure all users, but particularly non-SSO users, have a direct path from their user account profile page to security settings.

**Comment by ei8fdb** [comment]
> We could carve out time this upcoming year for a usability review of security settings

This sounds like a good way to solve this. My main question is will this design and development work be specific to OTF Hypha or up-stream in Hypha-core? Whichever is chosen please open it as an issue (and put it on the backlog) in the appropriate repo.

**Comment by frjo** [comment]
Everyone needs good 2FA functionality so of course this will be done so all can benefit.

**Comment by Techslammer** [comment]
> > We could carve out time this upcoming year for a usability review of security settings
> 
> This sounds like a good way to solve this. My main question is will this design and development work be specific to OTF Hypha or up-stream in Hypha-core? Whichever is chosen please open it as an issue (and put it on the backlog) in the appropriate repo.

I got you @ei8fdb. I'm going to spend time to define (temporarily) where issues like this should go when OTF staff bring it up and update next week--hopefully before maintainers call....Thumbs up to a usability review of security settings based on different roles. Look forward to talking about that.

**Comment by Techslammer** [comment]
@ei8fdb let's keep this in the OTF instance until OTF finalizes our processes internally. We (OTF) haven't set a priority level for this issue and there is no timeline set for it either which I'd like to do apart from the possible usability review of security settings. @fourthletter and I will continue to discuss (temporary) pathways for opening issues with the goal of working towards a first iteration of a permanent process. Hopefullly, by end of February.

**Comment by Erioldoesdesign** [comment]
> > We could carve out time this upcoming year for a usability review of security settings
> 
> This sounds like a good way to solve this. My main question is will this design and development work be specific to OTF Hypha or up-stream in Hypha-core? Whichever is chosen please open it as an issue (and put it on the backlog) in the appropriate repo.

I also understand this as a pan-adopter concern and also recognising this is waiting on OTF deciding on choices on how dev work happens e.g. on a hard fork or not hard fork etc.

Flagging my own opinion as a custodian of hypha's shared repo that powers all instances in that i would not want critical settings work to be made without at least acknowledgement and approval of other adopters. 

I think this is a governance complication though as yes, other adopters might not 'pay' or 'contribute' to this feature (though that could change) and would benefit from the improvements so i would ask: 
What would OTF want my role as product manager across adopters and for the shared, main hypha that powers each instance want me to facilitate and to not facilitate?

**Comment by Techslammer** [comment]
@Erioldoesdesign Thanks for the question.  I'll definitely ruminate over this especially since I think there is some language interpretation that i'd like to clarify within OTF and with other maintainers ex. I want to dig into more how we define a development fork versus a hard form when a dev is asked to work on a bug and/or feature on the OTF instance.

**Comment by frjo** [comment]
We should rename this issue or close it and create a new issue for the UX improvements.

**Comment by Techslammer** [comment]
It is unclear to me if I can't access 2fa settings because my role in the webapp is **staff** even though I am not using the sso option to sign into the webapp. What I do know is that OTF staff should be using sso so that it substitutes having to enable 2fa on their webapp account. I'm hoping all other roles can also access 2fa easily based on what you shared in the screenshot. 

@fourthletter what is your capcity to reproduce this issue on test to find which roles are having this issue. Then we can either close out this repo or rename it. The usability review of the security settings can be separate from at least identifying and fixing an issue that makes it difficult for an account holder no matter their role to enable, disable, and modify 2fa settings.

**Comment by Techslammer** [comment]
If this issue only affects staff that don't use sso, we can wait to address this through a usability review of security settings. If it affects other roles, than I have to explore another approach that includes a different timeline post the holidays.

**Comment by fourthletter** [comment]
Here is how a non-SSO user, e.g., applicants access their 2fa settings:

![Kapture 2FA Hypha](https://user-images.githubusercontent.com/20019656/147088843-5e850df1-cbe0-4349-b177-5a75918bf062.gif)

I agree it would be beneficial to have a usability review of security settings to improve the usability of the account section for all users. We could open a new epic issue sometime next year.

**Comment by Erioldoesdesign** [comment]
Suggest that until this is able to be put into any kind of scope of work we tag this as an 'improvement', perhaps with a label :)
